lib/r10k/action/deploy/environment.rb

```diff
@@ -43,6 +44,23 @@ def call
 
         private
 
+        def read_purge_allowlist (whitelist, allowlist)
+          whitelist_has_content = !whitelist.empty?
+          allowlist_has_content = !allowlist.empty?
+          case
+              when whitelist_has_content == false && allowlist_has_content == false
+                []
+              when whitelist_has_content && allowlist_has_content
+                raise R10K::Error "Values found for both purge_whitelist and purge_allowlist. Setting " <<
+                                  "purge_whitelist is deprecated, please move all values to purge-allowlist."
```

Comment:
same question as in the code-manager PR: are we sure it's preferable to disallow both keys, from an upgrading perspective?

Reply:
I've rejiggered the CM PR so this isn't an issue; it seems to me that if both are defined, than the upgrade hasn't been implemented correctly. I'm open to being more lax, but it seems more correct to error when two values are presented trying to configure the same thing.

Reply:
I'm with Tony, I don't think there's a reason to allow both simultaneously. We need to still support both keys, but should only allow you to have one or the other.

Reply:
okay